cli/operator/node.go

```diff
@@ -159,36 +187,100 @@ var StartNodeCmd = &cobra.Command{
 		}
 
 		cfg.SSVOptions.ValidatorOptions.DutyRoles = []spectypes.BeaconRole{spectypes.BNRoleAttester} // TODO could be better to set in other place
+
+		storageRoles := []spectypes.BeaconRole{
+			spectypes.BNRoleAttester,
+			spectypes.BNRoleProposer,
+			spectypes.BNRoleAggregator,
+			spectypes.BNRoleSyncCommittee,
+			spectypes.BNRoleSyncCommitteeContribution,
+			spectypes.BNRoleValidatorRegistration,
+		}
+		storageMap := ibftstorage.NewStores()
+
+		for _, storageRole := range storageRoles {
+			storageMap.Add(storageRole, ibftstorage.New(cfg.SSVOptions.ValidatorOptions.DB, storageRole.String(), cfg.SSVOptions.ValidatorOptions.ForkVersion))
+		}
+
+		cfg.SSVOptions.ValidatorOptions.StorageMap = storageMap
+		cfg.SSVOptions.ValidatorOptions.Metrics = metricsReporter
+
 		validatorCtrl := validator.NewController(logger, cfg.SSVOptions.ValidatorOptions)
 		cfg.SSVOptions.ValidatorController = validatorCtrl
+		cfg.SSVOptions.Metrics = metricsReporter
 
 		operatorNode = operator.New(logger, cfg.SSVOptions, slotTicker)
 
 		if cfg.MetricsAPIPort > 0 {
-			go startMetricsHandler(cmd.Context(), logger, db, cfg.MetricsAPIPort, cfg.EnableProfile)
+			go startMetricsHandler(cmd.Context(), logger, db, metricsReporter, cfg.MetricsAPIPort, cfg.EnableProfile)
 		}
 
-		nodeChecker.Wait()
+		nodeProber.Wait()
 
-		nodeUnreadyHandler := func() {
+		nodeProber.SetUnreadyHandler(func() {
 			logger.Fatal("Ethereum node(s) are either out of sync or down. Ensure the nodes are ready to resume.")
+		})
+
+		metricsReporter.SSVNodeHealthy()
+
```

Comment:
I would add comment that the code beneath is meant to handle events from execution client. or move all of it to a function that specifics it .

Reply:
Extracted to a function